src/js/game/hud/parts/waypoints.js

```diff
@@ -295,6 +284,59 @@ export class HUDWaypoints extends BaseHUDPart {
         this.rerenderWaypointList();
     }
 
+    /**
+     * Requests to edit a marker.
+     * @param {Waypoint} waypoint
+     */
+    requestEditMarker(waypoint) {
```

Comment:
Could you change it so that the same method for creating and editing a marker is used?

Reply:
https://github.com/tobspr/shapez.io/blob/2722366d686fd551a8edceebafc8e78c1e40e976/src/js/game/hud/parts/waypoints.js#L224-L227
How about this?